lineage/Lineage_utils.py

```diff
@@ -348,3 +349,81 @@ def gammaAnalytical(X):
+    """
+    In this function we remove the cells that are unfinished at the end and restrict our end-time analysis and build the model 
+    up to some time-point, which is intended_end_time. 
+    Here we first loop over the leaf cells and get the maximum tau of those, then to make sure we avoid unfinished cells in the
+    new population, we add it to 1 [hour] and then this will be the time-interval from the right (end of the experiment). In this
+    way we find the suitable end-time so that for all the cells in the lineage we have the tau and fate of all cells.
+    
+    Args:
+    -----
+        X (list of objects): a list holding the cells of the population as objects.
+        experimentTime (int/float): experiment time for simulation -- the same as before.
+    Returns:
+    --------
+        new_population (list of objects): after removing those cells at the end that we don't know their fate and end time.
+    *** Make sure you run the experiment long enough, experimentTime >>1 to have a reasonable number of cells at the end
+    """
+    
+    new_population = []
+    leaf_cell_taus = []
+
+    # first remove singleton lineages
+    X = remove_singleton_lineages(X)
+    
+    # get the lifetime of leaf cells and append them to a list
+    for cell in X:
+        if cell.isLeaf():
+            if cell.isUnfinished():
+                leaf_cell_taus.append(cell.tauFake)
+            else:
+                leaf_cell_taus.append(cell.tau)
+
+    # find the intended end of experiment time by maximum tau of leaf cells
+    intended_interval = max(leaf_cell_taus) + 0.01
+    intended_end_time = experimentTime - intended_interval
+
+    # lose the cells that were born after intended experiment end time
+    for cell in X:
+        if cell:
+            if cell.startT <= intended_end_time and not cell.isUnfinished():
+                # only if the cell is born before the intended experiment time 
+                # do we think about keeping the cell
+                if cell.isLeaf():
+                    # If the cell's start time is before our intended end time
+                    # and if the cell is a leaf
+                    # we don't have to do anything to it.
+                    pass 
+                elif not cell.isLeaf() and cell.endT > intended_end_time:
+                    # if the cell's start time is before the intended end time
+                    # and has daughter cells whose start times are after the intended
+                    # end time, then that parent cell is a leaf in our new population
+                    assert cell.left.startT > intended_end_time
+                    assert cell.right.startT > intended_end_time
+                    assert cell.left.startT == cell.right.startT
+                    cell.left = None
+                    cell.right = None
+
+                    assert cell.isLeaf() # new leaf being made 
+                elif not cell.isLeaf() and cell.endT <= intended_end_time:
```

Comment:
I'm pretty sure this section of code that I (I think I wrote IDK I forget) wrote is wrong. It doesn't make sense to make cells that start before the intended end time (cell.startT < intended_end_time) which is already implicit and cells that end before the intended end time (cell.endT <= intended_end_time) which is in the `if` condition into leaves because this comprises a vast majority of the cells in the tree. If anything this block of code should be taken out because nothing should be done to these cells.

Reply:
That is to say, lines 414-420 are wrong, I think and are not needed. Feel free to delete them and see if the tests work.

Reply:
Yes, agreed! we are removing those that should not be removed...